iocore/net/SSLNetVConnection.cc

```diff
@@ -834,7 +834,12 @@ void
 SSLNetVConnection::do_io_close(int lerrno)
 {
   if (this->ssl != nullptr && sslHandShakeComplete) {
```

Comment:
If the SSL handshake fails to complete, can it be the case that the `START` hook is called but not the `CLOSE` hook?

Reply:
Yes, as it stands that can be the case.  Probably not a great thing.  Will look into adding the hook invocation in the pre-handshake complete error cases.